include/iarc7_fc_comms/CommonFcComms.hpp

```diff
@@ -97,6 +101,12 @@ namespace FcComms{
             have_new_direction_command_message_ = true;
         }
 
+        inline void contactSwitchMessageHandler(
+                const iarc7_msgs::LandingGearContactsStamped::ConstPtr& message) {
+            last_contact_switch_message_ptr_ = message;
```

Comment:
A little research shows this should be right, it's being copy assigned, which releases the pointer previously held by `last_contact_switch_message_ptr_` and increases the reference count on what it's being assigned to.